lightning/src/ln/onchaintx.rs

```diff
@@ -966,7 +979,7 @@ impl<ChanSigner: ChannelKeys> OnchainTxHandler<ChanSigner> {
 				}
 			}
 		}
-		if self.prev_holder_commitment.is_some() {
+		if htlc_tx.is_none() && self.prev_holder_commitment.is_some() {
```

Comment:
I don't think we have test coverage for this code path ?

Reply:
it's covered by `claim_htlc_outputs.*` and some other functional tests.

Reply:
Yes covered after tweaking it 

```
failures:
    ln::functional_tests::test_bump_penalty_txn_on_revoked_htlcs
    ln::functional_tests::test_justice_tx
    ln::functional_tests::test_static_spendable_outputs_justice_tx_revoked_htlc_success_tx
    ln::functional_tests::test_static_spendable_outputs_justice_tx_revoked_htlc_timeout_tx
```